### PR TITLE
Maintain same LR schedule for early stop debug runs 

### PR DESCRIPTION
**Pull request by acisseJZhong**
addressed user request  #2326 to debug training runs with early stop steps but maintain the original lr schedule. 

**Test:**
```
[lr_scheduler]
warmup_steps = 2 
# Users need to explicitly add `schedule_steps` to overwrite; default `job_config.training.steps`.
total_steps = 1000
```
```
CONFIG_FILE="./torchtitan/models/qwen3/train_configs/qwen3_0.6b.toml" ./run_train.sh
```

### CODE REVIEW

File: torchtitan/config/job_config.py [hunk 1]
```diff
@@ -172,6 +172,14 @@ class LRScheduler:
     Steps for lr scheduler warmup, normally 1/5 of --training.steps
     """
 
+    schedule_steps: int | None = None
```
Comment (tianyu-l) [inline torchtitan/config/job_config.py]: `schedule_steps` in `Scheduler` sounds redundant, suggesting
```suggestion
    total_steps: int | None = None
```

File: torchtitan/train.py [hunk 2]
```diff
@@ -278,8 +278,13 @@ def __init__(self, job_config: JobConfig):
         self.optimizers = self.train_spec.build_optimizers_fn(
             self.model_parts, job_config.optimizer, parallel_dims, self.ft_manager
         )
+        lr_schedule_steps = (
```
Comment (tianyu-l) [inline torchtitan/train.py]: maybe put this in `build_lr_schedulers_fn` instead of train.py.

File: torchtitan/components/lr_scheduler.py [hunk 3]
```diff
@@ -104,28 +104,35 @@ def build_lr_schedulers(
         lr_scheduler_config (LRSchedulerConfig): The lr scheduler config.
         training_steps (int): The total number of training steps.
     """
+    # Use total_steps from config if set, otherwise fall back to training_steps
+    schedule_steps = (
```
Comment (tianyu-l) [inline torchtitan/components/lr_scheduler.py]: similarly, schedule steps in lr_scheduler.py sounds redundant, can we use total_steps? I think it's unambiguous in this file.

**Review by tianyu-l — APPROVED** [review]
one more nit comment

File: torchtitan/components/lr_scheduler.py (hunk 3 above)
Comment (acisseJZhong) [reply torchtitan/components/lr_scheduler.py]: sure, I personally think `schedule_steps` is more intuitive. But yeah it's better we unify to what's in the config.
